aptos-move/framework/supra-framework/sources/automation_registry_state.move

```diff
@@ -0,0 +1,345 @@
+    #[test(framework = @supra_framework)]
+    fun check_task_registration(framework: signer) acquires AutomationRegistryState {
+        initialize_registry_state_test(&framework);
+
+        let account = account::create_account_for_test(@0x123456);
+        register(&account,
+            vector[0, 1, 2, 3, 4],
+        100,
+        10,
+        20,
+        1,
+        vector[0, 1, 2 ,3],
+        );
+        assert!(1 == get_next_task_index(), 1);
+        assert!(10 == get_gas_committed_for_next_epoch(), 1)
+    }
+
+    #[test(framework = @supra_framework)]
+    #[expected_failure(abort_code = 2)]
+    fun check_registration_with_overflow_gas_limit(framework: signer) acquires AutomationRegistryState {
+        initialize_registry_state_test(&framework);
+
+        let account = account::create_account_for_test(@0x123456);
+        register(&account,
+            vector[0, 1, 2, 3, 4],
+            100,
+            70,
+            20,
+            1,
+            vector[0, 1, 2 ,3],
+        );
+        assert!(1 == get_next_task_index(), 1);
+        assert!(70 == get_gas_committed_for_next_epoch(), 1);
+        register(&account,
+            vector[0, 1, 2, 3, 4],
+            100,
+            70,
+            20,
+            1,
+            vector[0, 1, 2 ,3],
+        );
+    }
+
+    #[test(framework = @supra_framework)]
+    fun check_task_activation_on_new_epoch(framework: signer) acquires AutomationRegistryState {
+        initialize_registry_state_test(&framework);
+        let payload_tx = vector<u8>[0, 1, 2, 3, 4];
+        let txn_hash = vector<u8>[1, 2, 3, 4, 5];
+
+        let account = account::create_account_for_test(@0x123456);
+        register(&account,
+            payload_tx,
+            100,
+            10,
+            20,
+            1,
+            txn_hash,
+        );
+        // When moving to next epoch this task will be considered as expired
+        register(&account,
+            payload_tx,
+            25,
+            10,
+            20,
+            1,
+            txn_hash,
+        );
+        register(&account,
+            payload_tx,
+            150,
+            10,
+            20,
+            1,
+            txn_hash,
+        );
+        // When moving to next epoch this task will be considered as expired for the updcoming new epoch
+        register(&account,
+            payload_tx,
+            75,
+            10,
+            20,
+            1,
+            txn_hash,
+        );
+        // No active task and committed gas for the next epoch is total of the all registered tasks
+        assert!(40 == get_gas_committed_for_next_epoch(), 1);
+        let active_task_ids = get_active_task_ids();
+        assert!(active_task_ids == vector[], 1);
+
+        timestamp::update_global_time_for_test_secs(50);
+        on_new_epoch(30 * MICROSECS_CONVERSION_FACTOR);
+        // Committed gas for the next epoch only for 2 tasks 0 and 2, the task 3 will not be active during upcoming epoch
```

Comment:
I noticed that you are using the current module's register function for the unit test case.

However, the register function in the `automation_registry` module includes some additional assertion checks that are not being considered in this test.
I believe we should move those assertion checks to the `automation_registry_state` module to ensure consistency.

Reply:
agree, updated and added more unit tests.